torch/csrc/api/src/nn/modules/maxpool.cpp

```diff
@@ -0,0 +1,62 @@
+#include <torch/nn/modules/maxpool.h>
+
+#include <torch/expanding_array.h>
+
+namespace torch {
+namespace nn {
+
+template <size_t D, typename Derived>
+MaxPoolImpl<D, Derived>::MaxPoolImpl(MaxPoolOptions<D> options)
+    : options(std::move(options)) {}
+
+template <size_t D, typename Derived>
+void MaxPoolImpl<D, Derived>::reset() {}
+
+template <size_t D, typename Derived>
+void MaxPoolImpl<D, Derived>::pretty_print(std::ostream& stream) const {
+  stream << "torch::nn::MaxPool" << D << "d"
+         << "(kernel_size=" << options.kernel_size_
+         << ", stride=" << options.stride_ << ")";
+}
+
+Tensor MaxPool1dImpl::forward(const Tensor& input) {
+  return torch::max_pool1d(
+      input,
+      options.kernel_size_,
+      options.stride_,
+      options.padding_,
+      options.dilation_,
+      options.ceil_mode_);
```

Comment:
We might be missing `options.return_indices_` here (and the forward calls for 2d and 3d), I will push a commit to add it.

Reply:
Thanks. But I don't think that ```torch::max_poolxd``` uses it. That's why I didn't include it.

Reply:
Yes @yf225 here is the documentation: 
```
return_indices: if ``True``, will return the max indices along with the outputs.
                        Useful for :class:`torch.nn.MaxUnpool2d` later
```

Reply:
Ah got it, thanks a lot for the catch!